### PR TITLE
[docs] new JSON RPC API docs format

### PR DESCRIPTION
**Pull request by nickfrosty**
#### Problem

The JSON RPC API docs are difficult to navigate and consume. It should be easier to see the actual method docs and examples of how to use the method.

#### Changes

- created a new `/api` landing page (see picture below)
- creates a new display format in a "side-by-side" style format (picture below of what it looks like)
- each RPC method is segmented out into separate markdown "partial" files (currently living in `src/api`)
- the RPC methods for `http` and `websocket` methods have been moved to different pages
- custom sidebar for each of the `/api` pages to show only the methods related to the current viewed page (e.g. http or websocket methods)
- verified all recent changes of the JSON RPC page have been implemented into the newly formatted partials
  - verified commits affecting the JSON rpc page up until Jan 18, 2023 (which are all since this PR was opened)
  - history of changes [here](https://github.com/solana-labs/solana/commits/master/docs/src/developing/clients/jsonrpc-api.md)
- added a description of the `space` value returned for accounts to the API docs
  - see #28421

**NOTE:**
This updated display method will also pave the way to enable the core docs to have each of the client API docs on the same page, with a client "language" selector the user can use to view all the respective API docs for each client (e.g. JSON RPC via cURL, JavaScript, Rust)

#### Workflow for adding/updating RPC method doc pages

Making updates to the individual docs for each method will be inside of their respective "partial" MDX file, located in the respective `src/api` folder. See this example of the [`getAccountInfo` method doc](https://github.com/solana-labs/solana/pull/29772/files#diff-c8224a14a94942766b9289f6ae705b919a67c2c41d1afd2c3b40a2abf18bebba) for reference

Each of these markdown "partials" will be linked withing the corresponding api doc page (similar to importing a component in React, per the Docusaurus docs). Here is an example for the [`getAccountInfo` method](https://github.com/solana-labs/solana/pull/29772/files#diff-ddb6098b25a079fc81f66d6efc0042bef8fc44f82536538d9e1de710ff6cb3afR154-R156)

#### Screenshot of the proposed new format:

**API home page (located at url `/api`)**

Screenshot from 2022-10-22 08-46-57

**Example of the "side-by-side" view (this example is located at `/api/http#getaccountinfo`)**

Screenshot from 2022-10-22 08-40-04

**Read more "details" component expanded:**

Screenshot from 2022-10-22 08-58-08

### CODE REVIEW

File: docs/src/api/methods/_getBlockHeight.mdx [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+### Parameters:
+
+<Parameter type={"object"} optional={true}> 
+Configuration object containing the following fields:
+
+<Field
+  name="commitment"
+  type="string"
+  optional={true}
+  href="/api#configuring-state-commitment"
```
Comment (jacobcreech) [inline docs/src/api/methods/_getBlockHeight.mdx:30]: All of the `commitment` links need to point to `/api/http#configuring-state-commitment`

File: docs/src/api/deprecated/_getConfirmedBlock.mdx [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+  optional={true}
+>
+  level of transaction detail to return, either "full", "signatures", or "none"
+</Field>
+
+<Field name="rewards" type="bool" defaultValue={true} optional={true}>
+  whether to populate the `rewards` array.
+</Field>
+
+<Field name="encoding" type="string" defaultValue="json" optional={true} href="/api#parsed-responses">
```
Comment (jacobcreech) [inline docs/src/api/deprecated/_getConfirmedBlock.mdx:56]: All of the encoding fields need to point to `/api/http#parsed-responses`

File: docs/src/api/methods/_getHighestSnapshotSlot.mdx [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+## getHighestSnapshotSlot
+
+Returns the highest slot information that the node has snapshots for.
+
+This will find the highest full snapshot slot, and the highest incremental
+snapshot slot _based on_ the full snapshot slot, if there is one.
+
+:::caution
+NEW: This method is only available in solana-core v1.9 or newer. Please use
+[getSnapshotSlot](/api#getsnapshotslot) for solana-core v1.8
```
Comment (jacobcreech) [inline docs/src/api/methods/_getHighestSnapshotSlot.mdx:22]: This one needs to point to `/api/http#getsnapshotslot`

**Comment by jacobcreech** [comment]
There's a number of broken links on the page due to pointing to `/api` instead of `/api/http`. Are you not worried about the weird navigation going from `/developers`, clicking on `JSON RPC API` from the sidebar and being thrown into a new sidebar?

**Comment by nickfrosty** [comment]
@jacobcreech Incorrect links have been fixed. Including the final removal of the `jsonrpc-api.md` file and an outdated reference to an RPC method on the edge docs.
Also corrected the linking to the `filters` criteria object on 2 pages.

**Review by jacobcreech — APPROVED** [review]
Awesome work! In regards to the page switching on us between developers -> RPC, we should consider making RPC API its own thing as part of https://github.com/solana-labs/solana/issues/29744

**Review by jacobcreech — APPROVED** [review]
Thanks for the update! Looking forward to seeing how this does

**Comment by steviez** [comment]
Stumbled upon this today and leaving a note to say I'm a fan of the change!
image

**Pull request merged**